src/gameTest/java/mekanism/common/tests/energy/EnergyConversionTest.java

```diff
@@ -0,0 +1,237 @@
+              .thenExecute(() -> MekanismConfig.general.forgeConversionRate.set(defaultConversion))
+
+              // WRAPPING STRICT ENERGY TO FORGE ENERGY
+
+              //Test against a small empty container
+              .thenMap(() -> helper.createForgeWrappedStrictEnergyHandler(0, capacity))
+              .thenExecute(handler -> {
+                  int extracted = handler.extractEnergy(capacity, false);
+                  helper.assertValueEqual(extracted, 0, "extracted energy");
+                  int accepted = handler.receiveEnergy(capacity, false);
+                  helper.assertValueEqual(accepted, convertedCapacity, "accepted energy");
+                  helper.assertValueEqual(handler.getEnergyStored(), convertedCapacity, "stored energy");
+              })
+              //Test against a small full container
+              .thenMap(() -> helper.createForgeWrappedStrictEnergyHandler(capacity, capacity))
+              .thenExecute(handler -> {
+                  int accepted = handler.receiveEnergy(capacity, false);
+                  helper.assertValueEqual(accepted, 0, "accepted energy");
+                  int extracted = handler.extractEnergy(capacity, false);
+                  helper.assertValueEqual(extracted, convertedCapacity, "extracted energy");
+                  helper.assertValueEqual(handler.getEnergyStored(), 0, "stored energy");
+              })
+              //Test inserting against a small nearly full container
+              .thenMap(() -> helper.createForgeWrappedStrictEnergyHandler(capacity - 2, capacity))
+              .thenExecute(handler -> {//There shouldn't be any room for it
+                  int accepted = handler.receiveEnergy(capacity, false);
+                  helper.assertValueEqual(accepted, 0, "accepted energy");
+                  //Note: Needed energy should be 1 even though we can't accept it
+                  helper.assertValueEqual(handler.getEnergyStored(), convertedCapacity - 1, "stored energy");
+                  helper.assertValueEqual(handler.getMaxEnergyStored(), convertedCapacity, "max energy");
+              })
+              //Test extracting against a small nearly empty container
+              .thenMap(() -> helper.createForgeWrappedStrictEnergyHandler(2, capacity))
+              .thenExecute(handler -> {//There shouldn't be enough to get a single unit out
+                  int extracted = handler.extractEnergy(capacity, false);
+                  helper.assertValueEqual(extracted, 0, "extracted energy");
+              })
+              //Test inserting against a sub one sized container
+              .thenMap(() -> helper.createForgeWrappedStrictEnergyHandler(0, 2))
+              .thenExecute(handler -> {//There shouldn't be any room for it
+                  int accepted = handler.receiveEnergy(capacity, false);
+                  helper.assertValueEqual(accepted, 0, "accepted energy");
+              })
+              //Test against a small empty container
```

Comment:
test _what_ against an empty container. If 1RF is 2.5 Joules, shouldn't it store 2 Joules?